pkg/scheduler/scheduler.go

```diff
@@ -153,25 +164,20 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	for _, w := range workloads {
 		log := log.WithValues("queuedWorkload", klog.KObj(w.Obj), "clusterQueue", klog.KRef("", w.ClusterQueue))
 		cq := snap.ClusterQueues[w.ClusterQueue]
-		if cq == nil {
-			log.V(3).Info("ClusterQueue not found when calculating workload admission requirements")
-			continue
-		}
 		ns := corev1.Namespace{}
-		if err := s.client.Get(ctx, types.NamespacedName{Name: w.Obj.Namespace}, &ns); err != nil {
-			log.Error(err, "Looking up namespace")
-			continue
-		}
-		if !cq.NamespaceSelector.Matches(labels.Set(ns.Labels)) {
-			log.V(2).Info("Workload namespace doesn't match clusterQueue selector")
-			continue
-		}
 		e := entry{Info: w}
-		if !e.assignFlavors(log, snap.ResourceFlavors, cq) {
-			log.V(2).Info("Workload didn't fit in remaining clusterQueue even when borrowing")
-			continue
+		if cq == nil {
+			e.noAdmissionReason = "ClusterQueue not found when calculating workload admission requirements"
+		} else if err := s.client.Get(ctx, types.NamespacedName{Name: w.Obj.Namespace}, &ns); err != nil {
+			e.noAdmissionReason = err.Error()
```

Comment:
```suggestion
			e.noAdmissionReason = fmt.Sprintf("Could not obtain workload namespace: %v", err)
```

Reply:
the error message should include context, no?

Reply:
sure, but the user might be confused on why there is a message for a namespace failure

Reply:
not sure if saying "couldn't obtain workload namespace" removes that confusion, but I updated the message as suggested :)